File: sol/1353/1353.cpp
struct Solution {
    int maxEvents(vector<vector<int>>& events) {
        int n = events.size();
        ranges::sort(events, [](auto &a, auto &b){return a[0] < b[0];});
        vector<int> h;
        int res = 0, currDay = 1, ptr = 0;
        while (true){
            if (h.empty() && ptr < n){
                currDay = events[ptr][0];
            }
            while (ptr < n && events[ptr][0] <= currDay){
                h.push_back(-events[ptr++][1]);
                ranges::push_heap(h);
            }
            while (!h.empty() && -h.front() < currDay){
                ranges::pop_heap(h);
                h.pop_back();
            }
            if (!h.empty() && -h.front() >= currDay){
                ranges::pop_heap(h);
                h.pop_back();
                ++res;
                ++currDay;
            }
            if (ptr == n && h.empty()){
                break;
            }
        }
        return res;
    }
};

File: sol/1353/1353.py
import heapq as hq

class Solution:
    def maxEvents(self, events: List[List[int]]) -> int:
        events.sort(key=lambda itv: itv[0])
        h = []
        res = 0
        currDay = 1
        ptr = 0
        n = len(events)
        while True:
            if len(h) == 0 and ptr < n:
                currDay = events[ptr][0]
            while ptr < n and events[ptr][0] <= currDay:
                hq.heappush(h, events[ptr][1])
                ptr += 1
            while len(h) != 0 and h[0] < currDay:
                hq.heappop(h)
            if len(h) != 0 and h[0] >= currDay:
                hq.heappop(h)
                res += 1
                currDay += 1
            if ptr == n and len(h) == 0:
                break
        return res
